src/server/qgswmsserver.cpp

```diff
@@ -889,9 +884,10 @@ QImage* QgsWmsServer::getLegendGraphics()
 
   if ( contentBasedLegend )
   {
-    legendSettings.setMapScale( mMapRenderer->scale() );
-    double scaleFactor = mMapRenderer->outputUnits() == QgsMapRenderer::Millimeters ? mMapRenderer->outputDpi() / 25.4 : 1.0;
-    legendSettings.setMmPerMapUnit( 1 / ( mMapRenderer->mapUnitsPerPixel() * scaleFactor ) );
+    QgsMapSettings mapSettings;
+    legendSettings.setMapScale( mapSettings.scale() );
+    double scaleFactor = mConfigParser->outputUnits() == QgsUnitTypes::RenderUnit::RenderMillimeters ? mapSettings.outputDpi() / 25.4 : 1.0;
+    legendSettings.setMmPerMapUnit( 1 / ( mapSettings.mapUnitsPerPixel() * scaleFactor ) );
```

Comment:
Does this bit of code work? It looks like a temporary map settings instance is constructed to get configuration from it (which will be just some default values)